### ISSUE TITLE
Experience on M1 Mac

### ISSUE THREAD

**Issue by temp** [marcinguy/betterscan-ce#27]
Tried to run scanmycode locally on my M1 Macbook with docker-for-mac and stumbled onto some problems.

1. Port 5000 is used by the Mac Control Center by default - and changing it to 5001 via the exposed port in docker-compose.yaml is not carried through - every generated link still points to localhost:5000.
2. There is no pre-build arm64 image, and building it fails. When I change gosec from amd64 to arm64 it builds the images, and I can run the stack. But at runtime some errors about glibc are thrown.

**Comment by marcinguy** [comment]
Thank you for your feedback and ideas.

I don't have MacOS device. However, based on your feedback I made some changes

Ideally SMC should be run on Linux (Cloud Kubernetes, Server)


There is a "macos" branch with port being 5001 and disabled "confused" checker. It is a x86_64 binary, hence could cause problems on MacOS (arm64). Didn't have time to properly build it. This can be fixed. (could be the reason for your glibc errors)

Feel free to look around in the code/config files if you want to change anything else.

You can see

`dockermacos`

and 

`dockerhubmacos`

Built the images in `dockermacos` like this:

`backend`
`docker buildx build --platform linux/arm64 . -t scanmycode/scanmycode3-ce:server-arm64 --push`

`worker`
`docker buildx build --platform linux/arm64 . -t scanmycode/scanmycode3-ce:worker-arm64 --push`

`worker-cli`
`docker buildx build --platform linux/arm64 . -t scanmycode/scanmycode3-ce:worker-cli-arm64 --push`

Seems to have built clean.

Untested :)


If you wish to contribute/work on MacOS support or any contribution, please let me know.

Pulling images from `dockerhubmacos` should work

Feel free to make PR how to make it easier, if you know.

**Comment by madalinignisca** [comment]
Any chances to have the docker images built for arm64 linux also? It's not really about Mac with arm64, but there are also developers on arm64 working either with remote servers or just got recent Windows or Linux laptops/workstations with arm64.

PS: yes, adoption of arm64 on normal computing and mainly in servers is huge today, because of very low price compared to AMD64

**Comment by marcinguy** [comment]
Hi @madalinignisca Thank you for you message.

Understand. Yea, was also somewhat aware of it (performance gain, costs decrease)

It would be some effort to also provide arm64 images.

I see since latest update those checkers were added. I checked arm64 availability below:

kubescape - yes
insidersec - no
gr audit - yes
text4shell -yes
yara - yes
fluidattackscanner - yes
gostaticcheck - yea
gptanalyzer - yes
aquasec - yes

One does not have arm64 build (would have to be forked and build manually then - golang)

If anybody wants to take a stab at if, feel free to contact me.

Otherwise it would be a paid feature/consulting & deployment service. Feel free to reach out directly.
